### DISCUSSION TITLE
Is ValidatedForm.fetcher compatible with useTypedFetcher from remix-typedjson?

### DISCUSSION THREAD

**Discussion by ayrtonvwf** [airjp73/rvf#209] [Q&A]
I'm unsure if this is something I'm not doing straight.

`remix-validated-form` expects a `FetcherWithComponents<never>` on the `ValidatedForm.fetcher`, which is exacly what Remix returns from the normal `useFetcher` hook.

When using [remix-typedjson](https://github.com/kiliman/remix-typedjson) to get a better typing for the deserialization that happens on useFetcher (eg. `Date` fields coming from Prisma) we must use the `useTypedFetcher<typeof action>()` hook provided for the package instead, returning a `TypedFetcherWithComponents<something>`.

I've been working around this with the `as` hack from TypeScript.

```tsx
<ValidatedForm
  fetcher={commentSubmiter as FetcherWithComponents<never>}
>
  ...
</ValidatedForm>
```

- Is there already a better way to do this that I'm overseeing?
- Is there room for improvement here? (eg. if we consider that `remix-validated-form` is being too strict on the expected typing and could expect something more broad/generic).
- Is it `remix-typedjson` that _should_ return a more compatible typing for the whole Remix ecosystem instead?

**Comment by airjp73** [comment]
This is related to #205. The `fetcher` prop was added before remix added a typed version, so it's currently not working.
